src/components/DynamicDimension/DynamicDimension.js

```diff
@@ -1,72 +1,57 @@
-import React, { useEffect, useState } from 'react'
+import React from 'react'
 import PropTypes from 'prop-types'
+import i18n from '@dhis2/d2-i18n'
+import { useDataEngine } from '@dhis2/app-runtime'
 
-import { useDataQuery } from '@dhis2/app-runtime'
 import ItemSelector from './ItemSelector'
-
-const itemsByDimensionQuery = {
```

Comment:
Alternatively this can be kept in the components just adding the paging (and filters) parameters.